### PR TITLE
fix test_train_split function in evaluation/protocol

### PR DESCRIPTION
**Pull request by vindex10**
#### Related Issue(s)
Fixes: "Train test split overflow" #90

#### Description of Changes
```
diff --git a/ampligraph/evaluation/protocol.py b/ampligraph/evaluation/protocol.py
@@ -102,7 +102,7 @@ def train_test_split_no_unseen(X, test_size=5000, seed=0, allow_duplication=Fals
     tolerance = len(X) * 10
     while idx_test.shape[0] < test_size:
         i = rnd.randint(len(X))
-        if dict_subs[X[i, 0]] > 1 and dict_objs[X[i, 2]] > 1 and dict_rels[X[i,1]] > 1:
+        if dict_subs[X[i, 0]] > 0 and dict_objs[X[i, 2]] > 0 and dict_rels[X[i,1]] > 0:
             dict_subs[X[i, 0]] -= 1
             dict_objs[X[i, 2]] -= 1
             dict_rels[X[i, 1]] -= 1
```
and fixed tests.

### CODE REVIEW

**Comment by vindex10** [comment]
issue resolved. the condition is there for purpose. see #90

**Comment by chanlevan** [comment]
Yes, indeed. We try to support splitting the train/test datasets without duplication in the test set. However there are some cases it's not possible and the _allow_duplicates_ flag is there to create a test set with duplication triples.
